test/move_effect_spikes.c

```diff
@@ -0,0 +1,73 @@
+#include "global.h"
+#include "test_battle.h"
+
+SINGLE_BATTLE_TEST("Spikes deal damage on switchin")
+{
+    u32 amountUsed;
+
+    PARAMETRIZE { amountUsed = 1; }
+    PARAMETRIZE { amountUsed = 2; }
+    PARAMETRIZE { amountUsed = 3; }
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        u32 cnt;
+        for (cnt = 0; cnt < amountUsed; ++cnt) {
+            TURN { MOVE(player, MOVE_SPIKES); }
+        }
+        TURN { SWITCH(opponent, 1); }
+    } THEN {
+        switch (amountUsed) {
+        case 1:
+            EXPECT_DAMAGE_EQ(opponent->hp, (opponent->maxHP - opponent->maxHP / 8));
+            break;
+        case 2:
+            EXPECT_DAMAGE_EQ(opponent->hp, (opponent->maxHP - opponent->maxHP / 6));
+            break;
+        case 3:
+            EXPECT_DAMAGE_EQ(opponent->hp, (opponent->maxHP - opponent->maxHP / 4));
+            break;
+        }
+        EXPECT(gSideTimers[B_SIDE_OPPONENT].spikesAmount == amountUsed);
+    }
+}
+
+// FIXME: Test Gravity, Ingrain (From Baton Bass)
+SINGLE_BATTLE_TEST("Balloon prevents spike effect")
+{
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET) {Item(ITEM_AIR_BALLOON); };
+    } WHEN {
+        TURN { MOVE(player, MOVE_SPIKES); }
+        TURN { SWITCH(opponent, 1); }
+    } THEN {
+        EXPECT_DAMAGE_EQ(opponent->hp, opponent->maxHP);
+        EXPECT(gSideTimers[B_SIDE_OPPONENT].spikesAmount == 1);
+    }
+}
+
+// FIXME: Test Gravity, Ingrain (From Baton Bass)
+SINGLE_BATTLE_TEST("Type prevents spike effect")
+{
+    u32 item;
+    PARAMETRIZE { item = ITEM_NONE; }
+    PARAMETRIZE { item = ITEM_IRON_BALL; }
```

Comment:
I'd probably do parametrization of the damage here. Although it's a bit awkward because how do you handle "no damage" :thinking:

Perhaps instead of divisor I should have suggested a fixed-point number...
The first test could have had:
```c
u16 damageFraction; // UQ_4_12(1.0 / 8), UQ_4_12(1.0 / 6), UQ_4_12(1.0 / 4);
```
And this one would have:
```c
u16 damageFraction; // UQ_4_12(1.0 / 8), UQ_4_12(0.0)
```

Although, maybe the "we've already tested" thing applies and actually we just want `bool32 shouldDamage;`.
```c
    } THEN {
        if (shouldDamage)
            EXPECT_NE(opponent->hp, opponent->maxHP);
        else
            EXPECT_EQ(opponent->hp, opponent->maxHP);
    }
```

`EXPECT_NE` doesn't exist atm, but that's just because I haven't needed it yet.

Reply:
I refactored the test to match with the toxic spikes grounded checks and used
```
        if (shouldDamage)
            EXPECT(opponent->hp < opponent->maxHP);
        else
            EXPECT(opponent->hp == opponent->maxHP);
```